packages/scripts/scripts/format-js.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Node dependencies
+ */
+const { exit, stdout } = require( 'process' );
+
+/**
+ * External dependencies
+ */
+const chalk = require( 'chalk' );
+const { sync: spawn } = require( 'cross-spawn' );
+const { sync: resolveBin } = require( 'resolve-bin' );
+const { sync: dirGlob } = require( 'dir-glob' );
+const { sync: readPkgUp } = require( 'read-pkg-up' );
+
+/**
+ * Internal dependencies
+ */
+const {
+	fromConfigRoot,
+	fromProjectRoot,
+	getArgFromCLI,
+	getFileArgsFromCLI,
+	hasArgInCLI,
+	hasPackageProp,
+	hasProjectFile,
+} = require( '../utils' );
+
+// Check if the project has wp-prettier installed and if the project has a Prettier config
+function checkPrettier() {
+	try {
+		const prettierResolvePath = require.resolve( 'prettier' );
+		const prettierPackageJson = readPkgUp( { cwd: prettierResolvePath } );
+		const prettierPackageName = prettierPackageJson.pkg.name;
+
+		if ( ! [ 'wp-prettier', '@wordpress/prettier' ].includes( prettierPackageName ) ) {
+			return {
+				success: false,
+				message:
+					chalk.red( 'Incompatible version of Prettier was found in your project\n' ) +
+					"You need to install the 'wp-prettier' package to get " +
+					'code formatting compliant with the WordPress coding standards.\n\n',
+			};
+		}
+	} catch {
+		return {
+			success: false,
+			message:
+				chalk.red( "The 'prettier' package was not found in your project\n" ) +
+				"You need to install the 'wp-prettier' package under an alias to get " +
+				'code formatting compliant with the WordPress coding standards.\n\n',
+		};
+	}
+
+	// See: https://prettier.io/docs/en/configuration.html
+	const hasProjectPrettierConfig =
+		hasProjectFile( '.prettierrc' ) ||
+		hasProjectFile( '.prettierrc.json' ) ||
+		hasProjectFile( '.prettierrc.yaml' ) ||
+		hasProjectFile( '.prettierrc.yml' ) ||
+		hasProjectFile( '.prettierrc.js' ) ||
+		hasProjectFile( '.prettierrc.config.js' ) ||
+		hasProjectFile( '.prettierrc.toml' ) ||
+		hasPackageProp( 'prettier' );
+
+	if ( ! hasProjectPrettierConfig ) {
+		return {
```

Comment:
It's less pressing for the start but I would like to find the right path moving forward.


I find this part controversial given how all other scripts work. It should work perfectly fine without the config file defined in the root of the project. This is how it is handled elsewhere. I understand that it makes it tricky to work with IDEs but it gives the possibility to run it from CLI on demand, so it still allows some automation.

I'd love to see some hybrid approach (probably applicable also for linters and test runners). When the script runs for the first time, it informs that the IDE integration is missing and asks whether to set up one. If the answer is `yes` then the config is created in the roof of the project. What do you think?